great_expectations/cli/suite.py

```diff
@@ -79,10 +79,13 @@ def suite(ctx):
 @click.option(
     "--profile",
     "-p",
-    is_flag=True,
-    default=False,
-    help="""Generate a starting expectation suite automatically so you can refine it further. Assumes --interactive
-flag.
+    "profiler_name",
+    is_flag=False,
+    flag_value="",
+    default=None,
```

Comment:
This is somewhat going against Click patterns so the following approach is necessary.

We want to enable the following use-cases:

- `suite new` - No profiler involvement
- `suite new --profile` - Default to the legacy profiler
- `suite new --profile "my_profiler_name"` - Retrieve an RBP from store

`click.Option` can either be a flag (a boolean on/off switch) or an argument (which requires a value).

To ensure that we can treat this as both a boolean and traditional string arg, we need to use the following combination of values.

For the various options above, here's what happens to "profiler_name" in the arg list:

- `suite new` - Set to `None`
- `suite new --profile` - Set to `""`
- `suite new --profile "my_profiler_name"` - Set to `"my_profiler_name"`

Reply:
Thank you very much for this note, as well as the `help` message below 🙇🏼